pvdeg/standards.py

```diff
@@ -14,80 +14,181 @@
+        Meta data for a single location.
+
+    sky_model : str, optional
+        Options: 'isotropic', 'klucher', 'haydavies', 'reindl', 'king', 'perez'.
+    temp_model : str, optional
+        Options: 'sapm'.  'pvsyst' and 'faiman' will be added later.
+        Performs the calculation for the cell temperature.
+    conf_0 : str, optional        Model for the high temperature module on the exponential decay curve.
+        Default: 'insulated_back_glass_polymer'
+    conf_inf : str, optional
+        Model for the lowest temperature module on the exponential decay curve.
+        Default: 'open_rack_glass_polymer'
+    wind_factor : float, optional
+        Wind speed correction exponent to account for different wind speed measurement heights
+        between weather database (e.g. NSRDB) and the tempeature model (e.g. SAPM)
+        The NSRDB provides calculations at 2 m (i.e module height) but SAPM uses a 10 m height.
+        It is recommended that a power-law relationship between height and wind speed of 0.33 
+        be used. This results in a wind speed that is 1.7 times higher. It is acknowledged that 
+        this can vary significantly. 
+
+    Meta data
+    ------------------
+    tilt : float, 
+        Tilt angle of PV system relative to horizontal. [°] Required
+    azimuth : float,
+        Azimuth angle of PV system relative to north. [°] Required
+    Wind_Height_m : float,
+        if wind_factor is "None", it will run a calculation based on the height and a 
+        power factor of 0.33. If neither are supplied, the wind speed will not be adjusted.
+
+    Returns
+    -------
+    T_0 : float
+        An array of temperature values for a module with an insulated back or an 
+        alternatively desired small or zero standoff, [°C]. Used as the basis for the 
+        maximum achievable temperature.
+    T_inf : float
+        An array of temperature values for a module that is rack mounted, [°C].
+    T_measured : float
+        An array of values for the test module in the system, [°C] interest.
+    poa : float
+        An array of values for the plane of array irradiance, [W/m²]
+
+    """
+
+    parameters = ["temp_air", "wind_speed", "dhi", "ghi", "dni", "Module_Temperature"]
 
-def eff_gap(T_0, T_inf, level=1, T98=None, x_0=6.1):
+    if isinstance(weather_df, dd.DataFrame):
+        weather_df = weather_df[parameters].compute()
+        weather_df.set_index("time", inplace=True)
+    elif isinstance(weather_df, pd.DataFrame):
+        weather_df = weather_df[parameters]
+    elif weather_df is None:
+        weather_df, meta = weather.get(**weather_kwarg)
+
+    solar_position = spectral.solar_position(weather_df, meta)
+    poa = spectral.poa_irradiance(
+        weather_df,
+        meta,
+        sol_position=solar_position,
+        tilt=float(meta['Tilt']),
+        azimuth=float(meta['Azimuth']),
+        sky_model=sky_model, )
+    if 'Wind_Height_m' not in meta.keys():
+        wind_factor = 1
```

Comment:
we need to move this into weather, so we don't have to repeat it for all the functions

Reply:
The variable wind_factor is the exponential power law constant relating wind speed to height. It is not part of the meteorological data. The temperature.py now will use the wind speed values as is if someone tries to do module temperature calculations on a dataset that doesn't specify the wind speed measurement height. The assumption is that if someone provides their own data without the wind_speed_height specified it is because they know they are using the correct wind speed (at least we hope). 

However, in looking at the code, I put this fix in this location but then subsequently realized the wind speed should be corrected right before the temperature is calculated. These two lines of code are no longer effective because of a different check in temperature.py. Therefore, I'm just removing these two lines of code.

Reply:
okay